### PR TITLE
Improve translation test module

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | Read below
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | Read below

## What's this about

This PR replaces #13845 and solves the two first problems noted previously:

1. The test module `translation_test` has been renamed to `translationtest`, since translations cannot work on modules with underscores in their name. Indeed, the domain name for `translation_test` would be `Modules.Translationtest`, which can never be reversed back to `translation_test` to find the original module. [This required updating TranslationToolsBundle](https://github.com/PrestaShop/TranslationToolsBundle/pull/64).
2. The third component of a translation domain, in the case of modules, is the original file name. Many file names contain underscore, but it wasn't supported because the translation interface in BO used Inflector::tableize to separate the domain parts (yes domains are stored without dots in the translator dictionary, but that's another story). This has been fixed in [this commit](https://github.com/PrestaShop/PrestaShop/commit/9f1c2fe9f070a8952eff1bea8eb6ebbc229954d5)

In addition, this change also allows Module AdminControllers to reference admin templates (ie. templates located in `mymodule/views/templates/admin/`) without using the full route to the template ([see change](https://github.com/PrestaShop/PrestaShop/commit/504bbd558f895d6fb9a8d362afde5ac4f3a7112c)).

## How to test

1. Copy the `translationtest` module located in `tests/Resources/modules` over to the `/modules` folder. 
1. Go to modules catalog, search "translation" and install it.
1. Configure the module, you should see a simple page containing the wording "This wording belongs to the module file".
1. Go to `/admin-dev/index.php/modules/translations`. If your shop is in French you should get a screen with two wordings "Contrôleur moderne" and "Bonjour le monde"
1. Go to `/admin-dev/index.php?controller=AdminTranslationtestFoo`. If your shop is in French you should get a page containing the wording "Le template Smarty"
1. Go to `index.php?fc=module&module=translationtest&controller=bar`. If your shop is in French you should get a blank page containing only the wording "Ceci est un contrôleur frontal"

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15017)

### CODE REVIEW

File: src/PrestaShopBundle/Translation/Translator.php [hunk 1]
```diff
@@ -29,6 +29,9 @@
 
 use Symfony\Bundle\FrameworkBundle\Translation\Translator as BaseTranslator;
 
+/**
+ * Replaces the original Symfony FrameworkBundle translator
+ */
```
Comment (matthieu-rolland) [inline src/PrestaShopBundle/Translation/Translator.php:34]: I think you wanted to write "Replacement for the original Symfony FrameworkBundle translator" instead:

https://github.com/PrestaShop/PrestaShop/pull/14966#discussion_r310576557

File: tests/Resources/modules/translationtest/vendor/composer/ClassLoader.php [hunk 2]
```diff
@@ -279,7 +279,7 @@ public function isClassMapAuthoritative()
      */
     public function setApcuPrefix($apcuPrefix)
     {
-        $this->apcuPrefix = function_exists('apcu_fetch') && ini_get('apc.enabled') ? $apcuPrefix : null;
+        $this->apcuPrefix = function_exists('apcu_fetch') && filter_var(ini_get('apc.enabled'), FILTER_VALIDATE_BOOLEAN) ? $apcuPrefix : null;
```
Comment (matthieu-rolland) [inline tests/Resources/modules/translationtest/vendor/composer/ClassLoader.php:282]: Interesting way of checking a php config value (**FILTER_VALIDATE_BOOLEAN**), never saw it before 👍

Comment (eternoendless) [reply tests/Resources/modules/translationtest/vendor/composer/ClassLoader.php:282]: That was composer, not me, but interesting nonetheless!

File: src/PrestaShopBundle/Translation/Translator.php (hunk 1 above)
Comment (eternoendless) [reply src/PrestaShopBundle/Translation/Translator.php:34]: Right!

**Review by matthieu-rolland — APPROVED** [review]
LGTM

**Comment by jolelievre** [comment]
Thanks @eternoendless

**Pull request merged**